src/react-redux/reducer/apiclient-reducer.js

```diff
@@ -0,0 +1,140 @@
+import { findIndex } from 'lodash';
+import {
+	FRESH_DATA_CLIENT_ERROR,
+	FRESH_DATA_CLIENT_RECEIVED,
+} from '../action-types';
+
+const defaultState = {
+	endpoints: {},
+};
+
+const _reducers = {
+	[ FRESH_DATA_CLIENT_ERROR ]: reduceError,
+	[ FRESH_DATA_CLIENT_RECEIVED ]: reduceReceived,
+};
+
+/**
+ * Primary reducer for fresh-data apiclient data.
+ * @param {Object} state The base state for fresh-data.
+ * @param {Object} action Action object to be processed.
+ * @param {Object} [reducers] The mapping of reducers (used only for test).
+ * @return {Object} The new state, or the previous state if this action doesn't belong to fresh-data.
+ */
+export default function reducer( state = defaultState, action, reducers = _reducers ) {
+	const reducerFunc = reducers[ action.type ];
+	return reducerFunc ? reducerFunc( state, action ) : state;
+}
+
+/**
+ * Recursively reduce client endpoint state.
+ * 1. If it goes further down the path, recurse.
+ * 2. Otherwise if this has params, store it in queries.
+ * 3. Otherwise store it on this endpoint state.
+ * @param {Object} [state] Should contain endpoints property.
+ * @param {Object} action { apiName, clientKey, endpointPath, params (optional), data }
+ * @param {path} [path] The remaining path (used for recursion).
```

Comment:
should this be `{String}`?

Reply:
Oh, nope. But it should be `{Array}`. Good catch!